dill/_dill.py

```diff
@@ -483,8 +488,53 @@ def _restore_modules(unpickler, main_module):
         pass
 
 #NOTE: 06/03/15 renamed main_module to main
-def dump_session(filename='/tmp/session.pkl', main=None, byref=False, **kwds):
-    """pickle the current state of __main__ to a file"""
+def dump_session(
```

Comment:
This is nice. Two potentially controversial suggestions/thoughts:

1. rename `load_session` to something like `load_module`. Is there a better name for it now, given the current functionality?
2. rename `byref`, or integrate this `byref` with `settings['byref']`. It's confusing as is.

Reply:
Oh, great. I was trying to close the review I had opened in the beginning and ended up deleting all my last comments...

1. Agree. It's the best name and a way to change the API without breaking it.
2. Yes, but we need to choose one more name. Some ideas:

- `refer_imported`
- `disown_imported`
- `link_imported`
- `imported_byref`

Reply:
I like `dump_module` and `load_module`, but don't like `imported_byref`.  Any one word suggestions to replace `byref` here? How about `refimport`? `refmod`? `refmodule`?

Reply:
I do think the repeated use of `module` for many of the related functions is really nice.

Reply:
What about `refimported`? Because the option means:

> _Pickle all the imported objects by reference._

Even `refimport` makes me think of modules, and the option is not specific to modules.

By the way, I chose `imported_byref` semi-randomly to substitute `byref` in the last commit.

Reply:
`refimported` sound good to me

Reply:
As you said the other day, I don't love it, but is Good Enough™